sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -222,16 +222,20 @@ class HttpRequest(object):
     :param files: Files list.
     :param data: Body to be sent.
     :type data: bytes or str.
+    :keyword dict[str,any] query: A dictionary of query parameters you would like to include
+     in your request. We automatically format your query params into the inputted url.
     """
 
-    def __init__(self, method, url, headers=None, files=None, data=None):
-        # type: (str, str, Mapping[str, str], Any, Any) -> None
+    def __init__(self, method, url, headers=None, files=None, data=None, **kwargs):
+        # type: (str, str, Mapping[str, str], Any, Any, Any) -> None
         self.method = method
         self.url = url
         self.headers = _case_insensitive_dict(headers)
         self.files = files
         self.data = data
         self.multipart_mixed_info = None  # type: Optional[Tuple]
+        if "query" in kwargs:
```

Comment:
Let's just to a `pop` with a default value. No need to look it up twice :).

Reply:
If query is not set, we don't want to call self.format_parameters, right? (given self.format_parameters() has side effect even the input is None)